packages/reassure-compare/src/utils/validate.ts

```diff
@@ -0,0 +1,58 @@
+import { z } from 'zod';
+import { hasDuplicateValues } from './array';
+
+const performanceMetadataSchema = z.object({
```

Comment:
I think it's good idea to move the schemes and `z.infers` to `types` as they are more type related.